pkg/itests/labels_test.go

```diff
@@ -0,0 +1,95 @@
+package itests
+
+import (
+	"encoding/json"
+	"testing"
+
+	jsoningest "github.com/cloudprivacylabs/lsa/pkg/json"
+	"github.com/cloudprivacylabs/lsa/pkg/ls"
+)
+
+func TestProcessLabeledAs(t *testing.T) {
+	schStr := `
+	{
+		"@context": {
+			"ls":"https://lschema.org/"
+		},
+		"@type":"ls:Schema",
+		"@id": "http://testschema",
+		"https://lschema.org/layer": {
+			"@type": ["https://lschema.org/Object",
+					  "https://lschema.org/valueType"],
+			"@id": "root",
+			"ls:Object/attributes": [
+				{
+					"@id":  "attr1",
+					"@type": "ls:Value",
+					"labeledAs": "SOMELABEL"
+				},
+				{
+					"@id":  "attr2" ,
+					"@type": "ls:Value",
+					"ls:privacy": [
+						{
+							"@value": "flg1",
+							"labeledAs": "ANOTHERLABEL"
+						}
+					]
+				},
+				{
+					"@id":"attr3",
+					"@type": "ls:Value",
+					"ls:privacy": [
+						{"@value": "flg2"},
+						{"@value": "flg3"}
+					]
+				}
+			]
+		}
+	}
+	`
+
+	// inputStr := `{
+	// 	"field1": "value1",
+	// 	"field2": {
+	// 	   "t": "type1"
+	// 	}
+	// }`
+
+	var schMap interface{}
+	if err := json.Unmarshal([]byte(schStr), &schMap); err != nil {
+		t.Fatal(err)
+	}
+
+	schema, err := ls.UnmarshalLayer(schMap, nil)
+	if err != nil {
+		t.Error(err)
+	}
+
```

Comment:
labeledAs is for schemas only. So no doc ingestion during test. After unmarshalLayer, compile the schema, and check if the node has the label. For this to work, use ls:labeledAs in the schema